braindecode/datasets/tuh.py

```diff
@@ -123,21 +128,37 @@ def _parse_description_from_file_path(file_path):
     #                          subject/recording session/file
     # e.g.                 tuh_eeg/v1.1.0/edf/01_tcp_ar/027/00002729/
     #                          s001_2006_04_12/00002729_s001.edf
-    version = tokens[-7]
-    year, month, day = tokens[-2].split('_')[1:]
-    subject_id = tokens[-3]
+    if 'train' or 'eval' in tokens:
+        version = tokens[-6]
+    else:
+        version = tokens[-7]
+    subject_id = tokens[-1].split('_')[-2].split('s')[-1]
     session = tokens[-2].split('_')[0]
     segment = tokens[-1].split('_')[-1].split('.')[-2]
```

Comment:
```suggestion
    subject_id = tokens[-1].split('_')[0]
    session = tokens[-1].split('_')[1]
    segment = tokens[-1].split('_')[2].split('.')[0]
```
Fixes subject_id, session, and sample extraction from filename.

Reply:
Does not retain combability with older versions, needs to be added after dataset and version check if compatibility is desired